packages/modules/playwright/src/playwright-container.test.ts

```diff
@@ -0,0 +1,189 @@
+    const startedPlaywrightBuildInReporterContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    ).start();
+
+    const { output, exitCode } = await startedPlaywrightBuildInReporterContainer.exec([
+      "npx",
+      "playwright",
+      "test",
+      "--reporter=dot",
+    ]);
+
+    await startedPlaywrightBuildInReporterContainer.stop();
+
+    expect(output).toContain(SUCCESSFUL_TEST_RUNNING_OUTPUT);
+    expect(output).toContain(SUCCESSFUL_TEST_PASSED_OUTPUT);
+    expect(exitCode).toBe(SUCCESSFUL_TEST_EXIT_CODE);
+  });
+
+  it(`should pass example tests with a line build in reporter for ${process.arch}`, async () => {
+    const startedPlaywrightBuildInReporterContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    ).start();
+
+    const { output, exitCode } = await startedPlaywrightBuildInReporterContainer.exec([
+      "npx",
+      "playwright",
+      "test",
+      "--reporter=line",
+    ]);
+
+    await startedPlaywrightBuildInReporterContainer.stop();
+
+    expect(output).toContain(SUCCESSFUL_TEST_RUNNING_OUTPUT);
+    expect(output).toContain(SUCCESSFUL_TEST_PASSED_OUTPUT);
+    expect(exitCode).toBe(SUCCESSFUL_TEST_EXIT_CODE);
+  });
+
+  it(`should pass example tests creating an html reporter for ${process.arch}`, async () => {
+    const externalDestinationReporterPath = path.resolve(EXTERNAL_PLAYWRIGHT_SAVE_REPORTS_DIRECTORY, "index.html");
+
+    const startedPlaywrightContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    )
+      .withEnvironment({ PLAYWRIGHT_HTML_REPORT: "test-reports" })
+      .start();
+
+    const { output, exitCode } = await startedPlaywrightContainer.exec([
+      "npx",
+      "playwright",
+      "test",
+      "--reporter=html",
+    ]);
+
+    await startedPlaywrightContainer.saveReporter(
+      {
+        type: "html",
+        outputDir: "test-reports",
+      },
+      externalDestinationReporterPath
+    );
+
+    expect(output).toContain(SUCCESSFUL_TEST_RUNNING_OUTPUT);
+    expect(output).toContain(SUCCESSFUL_TEST_PASSED_OUTPUT);
+    expect(output).toContain("To open last HTML report run");
+    expect(output).toContain("npx playwright show-report");
+    expect(exitCode).toBe(SUCCESSFUL_TEST_EXIT_CODE);
+    expect(existsSync(externalDestinationReporterPath)).toBe(true);
+
+    await startedPlaywrightContainer.stop();
+  });
+
+  it(`should pass example tests creating a json reporter for ${process.arch}`, async () => {
+    const externalDestinationReporterPath = path.resolve(EXTERNAL_PLAYWRIGHT_SAVE_REPORTS_DIRECTORY, "results.json");
+
+    const startedPlaywrightContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    )
+      .withEnvironment({ PLAYWRIGHT_JSON_OUTPUT_NAME: "results.json" })
```

Comment:
same here than https://github.com/testcontainers/testcontainers-node/pull/726/files#r1525188563
do not able to redefine this envs and take the default playwright values

Reply:
refactor here https://github.com/testcontainers/testcontainers-node/pull/726/commits/b6930c480ce6836e15c7204eb7277432f84adeab